File: sharedse/src/main/java/com/ustadmobile/port/sharedse/impl/http/UmHttpResponseSe.java
package com.ustadmobile.port.sharedse.impl.http;

import com.ustadmobile.core.impl.http.UmHttpResponse;

import java.io.IOException;

import okhttp3.Response;

/**
 * Simple wrapper to wrap the OK HTTP library response object
 */

public class UmHttpResponseSe extends UmHttpResponse {

    private Response response;

    public UmHttpResponseSe(Response response) {
        this.response = response;
    }

    @Override
    public String getHeader(String headerName) {
        return response.header(headerName);
    }

    @Override
    public byte[] getResponseBody() throws IOException {
        return response.body().bytes();
    }

    @Override
    public boolean isSuccessful() {
        return response.isSuccessful();
    }
}

File: sharedse/src/main/java/com/ustadmobile/port/sharedse/impl/http/UmHttpResponseSe.kt
package com.ustadmobile.port.sharedse.impl.http

import com.ustadmobile.core.impl.http.UmHttpResponse

import java.io.IOException
import java.io.InputStream

import okhttp3.Response

/**
 * Simple wrapper to wrap the OK HTTP library response object
 */

class UmHttpResponseSe(private val response: Response) : UmHttpResponse() {

    override val responseBody: ByteArray?
        get() {
            try {
                return response.body()!!.bytes()
            } catch (e: IOException) {
                e.printStackTrace()
            }

            return null
        }

    override val responseAsStream: InputStream?
        get() = response.body()!!.byteStream()

    override val isSuccessful: Boolean
        get() = response.isSuccessful

    override val status: Int
        get() = response.code()

    override fun getHeader(headerName: String): String? {
        return response.header(headerName)
    }
}
